### ISSUE TITLE
Add alias support

### ISSUE THREAD

**Issue by ovalseven8** [irungentoo/toxcore#1590]
At the moment, every client implements an own system how to save aliases for contacts. Because every client should support aliases (they are really important because of how Tox works), it would makes sense to add it to the .tox file.

If you use client X with aliases and export the Tox profile because you'd like to use it on another device, everybody expect that the names in the contact list are the same as before.

Pros:
:white_check_mark: It will be much easier for client developers to implement alias support
:white_check_mark: Because of the consistent save system, import/export of a profile will be easier and more comfortable for the users
:white_check_mark: Global alias support and no client patchwork rug solution

**Comment by GrayHatter** [comment]
:-1:

**Comment by quininer** [comment]
also contact groups.

**Comment by GrayHatter** [comment]
> they are really important because of how Tox works

why are they important, i.e. how does tox work?

**Comment by ProMcTagonist** [comment]
@GrayHatter Usernames are not unique, or restricted in any meaningful way (which is how it should be). Aliases are therefore important to keep track of a contact across username changes, and to protect against impersonation.

**Comment by GrayHatter** [comment]
@ProMcTagonist you're correct, but with that you also explain why I don't want alias in toxcore*. Toxcore is meant to transfer (and to a very minor extent) store the information to and from your contacts. Unless you're also planning to send the alias to your friend. 

It doesn't belong in toxcore.

The exception to that being, if someone wants to extend the save format / functions built into toxcore. That would be okay, because then you could add or remove sections at will.

**Comment by ovalseven8** [comment]
At the moment, the names are already saved in the .tox file. It just makes sense to add also the aliases.

**Comment by GrayHatter** [comment]
no, names are in toxcore because toxcore provides an API to set and transfer names... aliases would be local only, that's the important difference

**Comment by ovalseven8** [comment]
I just wanted to make clear that because of that alias support would make sense.

**Comment by ProMcTagonist** [comment]
It sounds like aliases are in TCS domain, maybe depending on the expanded disguised zip save format that's been discussed

**Comment by GrayHatter** [comment]
As suggested in IRC, you could extend the toxcore save format by myself as well as @tux3. 

Then you could call tox_apis to set and retrieve that information in a cross compatible way, without having to implement anything like it in Messenger

**Comment by askmatey** [comment]
I do not see any reason to implement this feature into the core

**Comment by ace13** [comment]
One reason to implement it in core - or at least against core, would be to allow for syncing aliases between devices once multi device support is done.
Though that would probably still be more appropriate to do as a client-to-client protocol, not as part of the core.
